plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -126,17 +133,30 @@ public static ClientConfiguration buildConfiguration(Logger logger, Settings set
         return clientConfiguration;
     }
 
-    public static AWSCredentialsProvider buildCredentials(Logger logger, Settings settings, Settings repositorySettings) {
+    public static AWSCredentialsProvider buildCredentials(Logger logger, DeprecationLogger deprecationLogger,
+                                                          Settings settings, Settings repositorySettings) {
         AWSCredentialsProvider credentials;
         try (SecureString key = getValue(repositorySettings, settings,
                                          S3Repository.Repository.KEY_SETTING, S3Repository.Repositories.KEY_SETTING);
              SecureString secret = getValue(repositorySettings, settings,
                                             S3Repository.Repository.SECRET_SETTING, S3Repository.Repositories.SECRET_SETTING)) {
 
             if (key.length() == 0 && secret.length() == 0) {
-                // TODO: add deprecation, except for using instance profile
-                logger.debug("Using either environment variables, system properties or instance profile credentials");
-                credentials = new DefaultAWSCredentialsProviderChain();
+                // create a "manual" chain of providers here, so we can log deprecation of unsupported methods
+                AWSCredentials envCredentials = getDeprecatedCredentials(logger, deprecationLogger,
+                    new EnvironmentVariableCredentialsProvider(), "environment variables");
+                if (envCredentials != null) {
+                    credentials = new StaticCredentialsProvider(envCredentials);
+                } else {
+                    AWSCredentials syspropCredentials = getDeprecatedCredentials(logger, deprecationLogger,
+                        new SystemPropertiesCredentialsProvider(), "system properties");
+                    if (syspropCredentials != null) {
+                        credentials = new StaticCredentialsProvider(syspropCredentials);
+                    } else {
+                        logger.debug("Using instance profile credentials");
+                        credentials = new InstanceProfileCredentialsProvider();
+                    }
+                }
             } else {
                 logger.debug("Using basic key/secret credentials");
                 credentials = new StaticCredentialsProvider(new BasicAWSCredentials(key.toString(), secret.toString()));
```

Comment:
I don't see a reason to use a provider chain. We select exactly one set of credentials we are planning to use. I don't like having fallbacks (which a provider chain does, with its nasty blanket try/catch) when there are errors.